File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        var result = "Whatever."
        if (input.isBlank()) {
            result = "Fine. Be that way!"
        } else if (input.any(Char::isLetter) && input.uppercase() == input) {
            if (input.trim().last() == '?'){
                result = "Calm down, I know what I'm doing!"
            } else{
                result = "Whoa, chill out!"
            }
        } else if (input.trim().last() == '?') {
            result = "Sure."
        }
        return result
    }
}

File: haskell/bob/src/Bob.hs
module Bob (responseFor) where

import Data.Char
import Data.List

responseFor :: String -> String
responseFor xs
    | all isSpace xs = "Fine. Be that way!"
    | isQuestion = if isYelling then "Calm down, I know what I'm doing!"
                                else "Sure."
    | isYelling = "Whoa, chill out!"
    | otherwise = "Whatever."
    where isQuestion = last (dropWhileEnd isSpace xs) == '?'
          letters = filter isLetter xs
          isYelling = length letters /= 0 && all isUpper letters
